### PR TITLE
Allow libpki to verify quantum safe signatures

### PR DESCRIPTION
**Pull request by pi-314159**
Will create some new templates in oqs_template later

### CODE REVIEW

**Review by baentsch — COMMENTED** [review]
I'm a bit surprised to find all OQS sigalgs in a function called "IsAcceptableCurveForEcdsa": What's the rationale for this?

**Comment by pi-314159** [comment]
> I'm a bit surprised to find all OQS sigalgs in a function called "IsAcceptableCurveForEcdsa": What's the rationale for this?

Thank you for pointing this out! This was an old implementation that has been replaced by the change in `IsPublicKeyAcceptable`. I forgot to remove it :D

**Comment by pi-314159** [comment]
@baentsch Do you have any new suggestions? If no, I'll merge this PR and update OQS-Chromium build instructions in oqs-demos.

File: pki/simple_path_builder_delegate.cc [hunk 1]
```diff
@@ -123,6 +145,10 @@ bool SimplePathBuilderDelegate::IsPublicKeyAcceptable(EVP_PKEY *public_key,
     return true;
   }
 
+  if (IS_OQS_PKEY(pkey_id)) {
+    return true;
```
Comment (baentsch) [inline pki/simple_path_builder_delegate.cc:149]: This seems fishy. Any `pkey_id` is an OQS pkey?

**Review by baentsch — APPROVED** [review]
> @baentsch Do you have any new suggestions? If no, I'll merge this PR and update OQS-Chromium build instructions in oqs-demos.

Nope, just questions :) See above. But nothing that couldn't be fixed later on. Updating oqs-chromium instructions would be very welcome.

File: pki/simple_path_builder_delegate.cc (hunk 1 above)
Comment (pi-314159) [reply pki/simple_path_builder_delegate.cc:149]: pkey_id can be any type of pkey.

`IsPublicKeyAcceptable` will return false if the pkey is not an ECC, RSA, and quantum-safe key.

File: pki/verify_signed_data.cc [hunk 2]
```diff
@@ -249,6 +352,10 @@ bool VerifySignedData(SignatureAlgorithm algorithm, der::Input signed_data,
     }
   }
 
+  bool ret;
+  if (IS_OQS_PKEY(expected_pkey_id)) {
+    ret = oqs_verify_sig(public_key, signature_value_bytes.data(), signature_value_bytes.size(), signed_data.data(), signed_data.size()) ? true : false;
+  } else {
```
[the remark this answers is not in the document]
Comment (pi-314159) [reply pki/verify_signed_data.cc:358]: Yes. If the key is RSA or ECC, else will be executed

File: pki/simple_path_builder_delegate.cc (hunk 1 above)
Comment (baentsch) [reply pki/simple_path_builder_delegate.cc:149]: Sincere apologies. I completely mis-read the code. Time for the weekend. Please go ahead and merge!

**Pull request merged**